src/app/features/auth/services/auth.service.ts

```diff
@@ -1,42 +1,58 @@
-import { inject, Injectable, signal, WritableSignal } from '@angular/core';
-import { LocalStorageService } from '@core/services/local-storage.service';
+import { inject, Injectable, signal } from '@angular/core';
+import { Router } from '@angular/router';
+import { AlertService } from '@core/services/alert/alert.service';
+import { LocalStorageService } from '@core/services/local-storage/local-storage.service';
+import { UserRequest } from '@features/auth/interfaces/user-request.interface';
+import { LocalStorageKey } from '@shared/enums/local-storage-key.enum';
+import { catchError, EMPTY, tap } from 'rxjs';
 import { AuthApiService } from './auth-api.service';
 
-const KEY_USER_TOKEN = 'userToken';
-
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService {
+  private readonly router = inject(Router);
+
+  private readonly alerts = inject(AlertService);
+
   private authApiService = inject(AuthApiService);
 
-  private localStorage: LocalStorageService = inject(LocalStorageService);
+  private localStorage = inject(LocalStorageService);
 
-  public isLoggedIn: WritableSignal<boolean>;
+  private _isLoggedIn = signal(this.getAuthStatus());
 
-  public isAdminIn = signal(false);
+  public isLoggedIn = this._isLoggedIn.asReadonly();
 
-  constructor() {
-    this.isLoggedIn = signal(this.getAuthStatus());
-  }
+  private _isAdminIn = signal(false);
```

Comment:
Maybe it would be better to name the field just 'isAdmin' instead of 'isAdminIn'?

Reply:
this is an intermediate result, in the future it can replace the role